File: app/src/main/java/com/hap/popularmovie/model/movie/MovieResponse.java
package com.hap.popularmovie.model.movie;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

/**
 * Created by luis on 9/16/17.
 */

public class MovieResponse {
    private int page;
    @SerializedName("total_results")
    private int totalResults;
    @SerializedName("total_pages")
    private int totalPages;
    @SerializedName("results")
    private ArrayList<MovieItem> movies;

    public int getPage() {
        return page;
    }

    public int getTotalResults() {
        return totalResults;
    }

    public int getTotalPages() {
        return totalPages;
    }

    public ArrayList<MovieItem> getMovies() {
        return movies;
    }
}

File: app/src/main/java/com/hap/popularmovie/model/MovieResponse.kt
package com.hap.popularmovie.model

import com.google.gson.annotations.SerializedName
import java.util.*

/**
 * Created by luis on 10/4/17.
 */
// TODO 04 - Data classes
class MovieResponse(private val page: Int = 0,
                    @SerializedName("total_results")
                    private val totalResults: Int = 0,
                    @SerializedName("total_pages")
                    private val totalPages: Int = 0,
                    @SerializedName("results")
                    val movies: ArrayList<MovieItem>? = null)